### ISSUE TITLE
Go to definition or peek definition doesn't work when using 'using namespace' in the cpp file

### ISSUE THREAD

**Issue by darbitman** [microsoft/vscode-cpptools#4018]
Type: General

**Describe the bug**
- OS and Version: Ubuntu 18.04 x64
- VS Code Version: 1.36.1
- C/C++ Extension Version: 0.24.1

**To Reproduce**
1. Generate an hpp for some class Foo and place this in a named namespace. Call it 'test'
2. In the cpp file, include the header and also 'using namespace test' and define the methods without namespace resolution
3. When clicking on Go to definition or Peek definition in the hpp, it will not find the definition
4. However, when the definitions are wrapped up in the namespace itself the definitions will be found. Also if using namespace resolution they will be found as well.

Foo.hpp:
```
#ifndef FOO_HPP
#define FOO_HPP

namespace test
{
class Foo
{
    Foo();
    ~Foo();

    void DoSomething();
};
}  // namespace test

#endif
```

Foo.cpp (broken):
```
#include "Foo.hpp"

using namespace test;

Foo::Foo() {
    // do nothing
}

Foo::~Foo() {
    // do nothing
}

void Foo::DoSomething() {
    // do nothing
}
```

Foo.cpp (working 1):
```
#include "Foo.hpp"

test::Foo::Foo() {
    // do nothing
}

test::Foo::~Foo() {
    // do nothing
}

void test::Foo::DoSomething() {
    // do nothing
}
```

Foo.cpp (working 2):
```
#include "Foo.hpp"

namespace test {
Foo::Foo() {
    // do nothing
}

Foo::~Foo() {
    // do nothing
}

void Foo::DoSomething() {
    // do nothing
}
}
```

**Comment by Colengms** [comment]
Hi @darbitman .  Thanks for reporting this.  I can repro this on Linux with clang IntelliSense.

We share a lot of this code with VS.  I can't repro in a recent build of VS, using windows-clang-x64 IntelliSense mode.  It's possible this may be fixed in VS and we will pick up the fix when we next integrate code from VS.

**Comment by sean-mcmanus** [comment]
I believe this is bug https://github.com/microsoft/vscode-cpptools/issues/3902 that is fixed with our pending insiders release.

@Colengms Did you try the repro with our latest internal build?

**Comment by Colengms** [comment]
Yes, this appears to be the same issue and fixed in the latest pending release.

**Comment by sean-mcmanus** [comment]
Should be fixed with 0.25.0-insiders: https://github.com/microsoft/vscode-cpptools/releases/tag/0.25.0-insiders .
